calculator.js

```diff
@@ -1,28 +1,278 @@
+// TO:DO:
 
-const exampleAdditionInput = {
-  num1: 3,
-  num2: 5,
-  operation: 'add',
+// This program only allows for single set of parentheses
+// (3+5) --> OK
+// (3+5)*(6+3) --> not covered
+
+// This program does NOT allow more than one operation in an expression
+// No order of operations followed
+
+// This program doesn't allow expressions without enclosing parens
+// (3+5) --> OK
+// 3+5 --> NO
+
+//////////////////////////////////////////////////
+
+const validParens = function (num) {
+  let parens = [];
+  for (const char of num) {
+    if (char === '(') {
+      parens.push(char);
+    } else if (char === ')') {
+      if (parens.pop() !== '(') {
+        return false;
+      }
+    }
+  }
+
+  if (parens.length > 0) {
+    return false;
+  }
+  
+  return true;
+};
+
+const reduceParensExpression = function(num) {
+  num = num.replace('(', '');
+  num = num.replace(')', '');
+  
+  for (op in validOps) {
+    if (num.includes(op)) {
+      const tempInput = num.split(op);
+      const tempNum1 = tempInput[0];
+      const tempNum2 = tempInput[1];
+      const tempOp = op;
+      const tempSoln = computeWithValidations(tempNum1, tempNum2, tempOp); 
+      return tempSoln;
+    }
+  }
+
+  return num;
 }
 
+const evalParens = function (num, origNum, errors) {
+  if (!validParens(num)) {
+    error = `INVALID PARENS: ${origNum} does not use valid parentheses.`;
+    console.log(error);
+    errors += error;
+  } else {
+    numReduced = reduceParensExpression(num);
+    if (numReduced.soln || numReduced.soln === 0) {
+      num = numReduced.soln;
+      errors += numReduced.errors;
+    } else {
+      num = numReduced;
+    }
+  }
+
+  return { num, errors };
+};
+
+const validOps = {
+  add: '+',
+  '+': 'add',
+  subtract: '-',
+  '-': 'subtract',
+  exponent: ['^', '**'],
+  '^': 'exponent',
+  '**': 'exponent',
+  multiply: '*',
+  '*': 'multiply',
+  divide: '/',
+  '/': 'divide',
+  modulo: '%',
+  '%': 'modulo'
+};
+
+const validDivision = function (num, operator) {
+  if (num === 0 && (operator === '/' ||
+                    operator === 'divide' ||
+                    operator === '%' ||
+                    operator === 'modulo')) {
+    return false;
+  }
+
+  return true;
+};
+
+const validNum = function (num, errors) {
+  const temp = Number(num);
+  
+  if (typeof(temp) != 'number') {
+    return false;
+  } else if (Number.isNaN(temp)) {
+    return false;
+  } else if (temp === 0 && temp == num && num !== '0' && num !== 0) {
+    return false;
+  }
+
+  return true;
+};
+
+const computeExpression = function (num1, num2, operator, soln) {
```

Comment:
👍 Nice function  However you probably don't need the `soln` parameter. Just using a local variable is fine.